plugins/modules/gitlab_project_variable.py

```diff
@@ -354,14 +353,13 @@ def native_python_main(this_gitlab, purge, requested_variables, state, module):
         if purge:
             # refetch and filter
             gitlab_keys = this_gitlab.list_all_project_variables()
-            existing_variables = [x.attributes for x in gitlab_keys]
-            for item in existing_variables:
-                item.pop('project_id')
+            existing_variables = filter_returned_variables(gitlab_keys)
 
             remove = [x for x in existing_variables if x not in requested_variables]
             for item in remove:
-                if this_gitlab.delete_variable(item):
-                    return_value['removed'].append(item)
+                if item not in add_or_update:
```

Comment:
Shouldn't you also check if this is an unmodified variable?

And is this really necessary, and not already handled by `remove = [x for x in existing_variables if x not in requested_variables]`?

Reply:
Ah yes. this was a modification of the first draft. This must be reverted.